app/experimenter/normandy/tasks.py

```diff
@@ -152,15 +154,43 @@ def set_is_paused_value_task(experiment_id, recipe_data):
     metrics.incr("set_is_paused_value.completed")
 
 
-def update_population_percent(experiment, recipe_data):
+def update_population_info(experiment, recipe_data):
     if recipe_data and "filter_object" in recipe_data:
         filter_objects = {f["type"]: f for f in recipe_data["filter_object"]}
-        if "bucketSample" in filter_objects:
-            bucket_sample = filter_objects["bucketSample"]
-            experiment.population_percent = decimal.Decimal(
-                bucket_sample["count"] / bucket_sample["total"] * 100
+
+        update_population_percent(experiment, recipe_data, filter_objects)
+        update_firefox_versions(experiment, recipe_data, filter_objects)
+
+
+def update_population_percent(experiment, recipe_data, filter_objects):
+
+    if "bucketSample" in filter_objects:
+        bucket_sample = filter_objects["bucketSample"]
+        experiment.population_percent = decimal.Decimal(
+            bucket_sample["count"] / bucket_sample["total"] * 100
+        )
+        experiment.save()
+
+
+def update_firefox_versions(experiment, recipe_data, filter_objects):
+    changed_data = {}
+
+    if "version" in filter_objects:
+        versions = filter_objects["version"]
+        min_version = min(versions["versions"])
+        max_version = max(versions["versions"])
+        if experiment.firefox_min_version != min_version:
+            changed_data["firefox_min_version"] = min_version
+        if experiment.firefox_max_version != max(versions):
```

Comment:
dupe of line 181?

Reply:
Actually how does this even work, if `versions` is a dict and the actual version list is in `versions["versions"]`?